src/persist-txn/src/txns.rs

```diff
@@ -497,22 +490,12 @@ where
                 }
             };
 
-            // We know above that we've applied the latest write, so go ahead and
-            // fill the time between that and forget_ts with empty updates. It's
-            // fine if we get interrupted between the forget CaA and here, we just
-            // won't get around to telling the caller that it's safe to use this
-            // shard directly again.
             for data_id in registered.iter() {
```

Comment:
Looks like we do `data_write.remove` and `apply_le` in a different order in forget vs forget_all? I'm not sure if it matters, but I'd say standardize on one ordering for both. I think at the moment, it's probably most consistent with our idempotency story to do `data_write.remove` before `apply_le` (though maybe that changes when we support only applying a specific data shard)

Reply:
Yep, good call out. I just pushed a fix.